decidim/decidim
# Add script to fix faulty migrations

#### :tophat: What? Why?
This PR adds a task that aims to fix all the legacy migrations in an application. We discovered that when we are running pipelines on a legacy project ( in our case nightly) all the existing migrations are failing due to default_scoping. 

#### :pushpin: Related Issues
*Link your PR to an issue*
- Related to #13297
- Related to #13668

#### Testing
1. Reset your decidim codebase to a hash like `7c31dcf1b161e254664bad9888d2eed87a71f35e`
2. Generate a development / test application 
4. Update your decidim codebase to the latest version 
5. run `./bin/rails decidim:upgrade`
6. Reset your database ( `./bin/rails db:drop db:create` )
7. Run `./bin/rails db:migrate`
8. See the migration script is failing with missing column `deleted_at` on various files 
9. Apply this patch 
10. run `./bin/rails decidim:upgrade`
11. Run `./bin/rails db:migrate`
12. See everything works as intended 


:hearts: Thank you!
